Hints.py

```diff
@@ -1838,7 +1838,30 @@ def build_misc_location_hints(world: World, messages: list[Message]) -> None:
                     text = data['location_text'].format(item=get_hint(get_item_generic_name(item),
                                                                         world.settings.clearer_hints).text)
 
-        update_message_by_id(messages, data['id'], str(GossipText(text, ['Green'], prefix='')), 0x23)
+                update_message_by_id(messages, data['id'], str(GossipText(text, ['Green'], prefix='')), data['text_style'], allow_duplicates=True)
+            # Allows for other misc hints to use 'location_fallback, but avoids the mask hints from being overwritten.
+            if hint_type not in ['skull_mask', 'mask_of_truth'] and hint_type not in world.settings.misc_hints:
+                update_message_by_id(messages, data['id'], str(GossipText(text, ['Green'], prefix='')), data['text_style'], allow_duplicates=True)
+
+def build_misc_dual_hints(world: World, messages: list[Message]) -> None:
+    for hint_type, data in misc_dual_hint_table.items():
+        # Change misc_hints list to use the Skull Mask and Mask of Truth Dual hint
+        if 'skull_mask' in world.settings.misc_hints and 'mask_of_truth' in world.settings.misc_hints:
+            world.settings.misc_hints.remove('skull_mask')
+            world.settings.misc_hints.remove('mask_of_truth')
+            world.settings.misc_hints.append('mask_shop_dual')
```

Comment:
Mutating the `settings` object should be avoided, it can cause issues since there are some parts of the codebase that assume it remains static. Maybe you could make this less hardcoded and check for corresponding single/dual misc. location hints where relevant?